### ISSUE TITLE
Proposal - Appeal committee

### ISSUE THREAD

**Issue by trust1995** [code-423n4/org#73]
I believe that in order to further democratize the C4 experience it is necessary to add an additional stage to each contest - appeal committee. Wardens that are convinced their submission has been incorrectly judged, **after** Judging QA, should be able to file an appeal.

An appeal committee will be comprised of 3 judges, who did not judge the contest in question. By majority vote, they will come to a decision regarding a change in severity or re-opening of closed submission.

Appeals cannot be filed recklessly. They will require a deposit of $200-400, which will only be returned to the warden if the appeal is accepted. In this format, I don't expect appeals to be sent except in situations where they are justified, respecting the time value of judges.

Appeals will give further confidence and motivation to wardens, as they are guaranteed that their submissions would be judged correctly and in a well-balanced, nuanced tone. It is an inevitable step in the standardization trend currently in motion.

Clearly, the appeal stage will delay the completion of contests by several days. Contest pipelines are currently ~2 months long. I think it is a reasonable price to pay for the preservation of democracy in C4.

**Comment by Picodes** [comment]
Your idea is interesting!
 - from what I understood the goal especially with presort and the new judging extension is to reduce the post contest pipeline to 1 to 2 weeks, so it'd be better to find a way to minimize the back and forth and have QA + appeals settled in a few days
 - the deposit idea was also in the pipe for all findings at some point (for every submission there would be a small deposit to reduce spam)

**Comment by trust1995** [comment]
Yeah, the idea is definitely in the spirit of Alex's proposal to charge a refundable $5 submission fee.

**Comment by dmvt** [comment]
I like this idea, but as @Picodes said, I think this period should exist in tandem with the QA procedure or at most one day longer.

Proposed flow:
- During Post judging QA warden asks for clarity of reconsideration
- Judge responds
- Warden can appeal the response within 6 hour with a 500 USDC deposit
- 3 judge panel makes final decision as a 2 of 3
- Warden's deposit goes to the original judge if ruled against

**Comment by SmilingHeretic** [comment]
I like the idea of the appeal committee, I do think that we need a mechanism to prevent spam appealing, and I think that deposits would prevent spam appealing.

What I don't like about the idea of $200 - 400 deposits is that it effectively makes appeals not accessible for talented wardens from not very rich countries who haven't won large bug bounties yet. 

Giving advantage (backstage+ access and appeals in post judging QA) to wardens who can pass KYC already feels a bit unfair. Now you gain advantage for being rich from the beginning.

**Comment by trust1995** [comment]
@SmilingHeretic I understand this point of view, but still think that money is the best way to filter out the spam. Again, appeals are not an everyday measure, I wouldn't file it for under 90% of getting accepted. With that consideration, you don't necessarily need to be rich to use this measure.

**Comment by Picodes** [comment]
Also something not tackled by the proposal is that it'd require 3 judges to answer within 24-48h, so we may want to incentivize them somehow

**Comment by SmilingHeretic** [comment]
@trust1995 Well, if you simply don't have or can't afford to lose 500 USD, then 5-10% risk of the appeal not being accepted might be still too high.

On the other hand, if earning 500 USD without appeals is relatively easy for a talented beginner warden, then deposits might be an acceptable solution... I'm not sure.

Still, it'll feel more fair if we have some process for a person in situation "I want to appeal, I'm confident that my appeal will get accepted but I can't afford to lose 500 USD".

**Comment by trust1995** [comment]
@SmilingHeretic Another option to consider is to keep a negative balance of $500 for that warden, which will be cancelled with any future earnings. Still need to iron out the details on that.

@Picodes Maybe, if appeal was accepted, 300$ would go from judge awards to the 3 committee members. If appeal is rejected, $300 from deposit would go to committee, and $200 to the judge awards? It's important that committee has no economic incentive to rule either way.

**Comment by SmilingHeretic** [comment]
@trust1995 The issue with negative balance is that someone can abuse it and simply create another warden account for future contests.

It's similar to what [Sherlock does](https://docs.sherlock.xyz/audits/watsons/escalation-period):

> The 10 USDC is removed from the Watson's payout in that contest or the next contest where they receive a payout.

**Comment by berndartmueller** [comment]
I think the amount of $ required for an appeal has to be on the higher side to prevent spam. 

Wardens who are early in their career could try to get a "sponsor" who deposits for the warden. In case the appeal is accepted, the sponsor receives the deposit and the warden's submission is probably higher rewarded. Those additional rewards could then be used to pay interest to the deposit sponsor.

**Comment by dmvt** [comment]
> Also something not tackled by the proposal is that it'd require 3 judges to answer within 24-48h, so we may want to incentivize them somehow

I'm not sure this is needed. I would be happy to perform this duty for the good of the org. Perhaps we could award in bonus points for the judge scoring mentioned [here](https://github.com/code-423n4/org/issues/71).

> I think the amount of $ required for an appeal has to be on the higher side to prevent spam.

I agree wholeheartedly with this. A warden who chooses to appeal has already been through multiple rounds of conversations with a single judge. It's too easy to appeal on the off chance that they might get more money if they win.

> Wardens who are early in their career could try to get a "sponsor" who deposits for the warden. In case the appeal is accepted, the sponsor receives the deposit, and the warden's submission is probably higher rewarded. Those additional rewards could then be used to pay interest to the deposit sponsor.

I would love to see this economic sponsor be another warden in the contest. The funds at risk could come from their winnings. That way, the sponsoring warden has such a high conviction level that they know they will lose money in either scenario. Either the appeal is granted, and their winnings decrease slightly, or the appeal is lost, and their funds decrease.

**Comment by carlitox477** [comment]
What about a suspension for lack of proof as certified for a month?
For instance, I have made to appellations, one was resolved partially in my favor, the other one I've not got a response yet. I think that in both I've shared proof that it is worthy to read and take into account for a final judgment decision. Even if the last appellation does not resolve in my favour, I consider that I made clear arguments on why my appellation should have been consider valid.

Then I see other claims which just state a few lines, without almost proof on why a judges should reconsider their decisions.

I would suggest 2 better approaches to save judges time and improve resolution process:
* Option A: Select 2-3 volunteer wardens based on:
     1. Previous appellation solved in their favour in a 365 day frame: 35%
     2. Current rank in leader board in a 90 days frame: 35%
     3. Previous penalties received for bad conduct: 30%
  Then, separating max(300 USD for each warden; 1% of the contest pot) to give as reward for the volunteer wardens
  The function of this wardens is deciding whether an appellation worth the time of the judge or not to reconsider their decision.
* Option B: Penalize those certified warden who made appellation without any coherent proof with a 100 - 200 USD fine to by paid with future bounties, and a suspension of at least 1 month of certified privileges (once the month is achieved, they could decide to pay with their own money and not with future rewards).  About the fee, this is at discretion of judges if they consider the warden presented almost no proof to make even an argument to review their decision.

Option A will save judges time, involves promising warden who could eventually become judges to start taking part in judging process and understand it, but it will cost money from the reward pot.
Option B may or may not save time to judges, won't cost money from the reward pot, will kick out unprofessional certified wardens, and will allow judges to earn more money (when the fine is paid)

**Comment by JeffCX** [comment]
Shall we please implement this appeal process?

I would add a few points:

- the warden (appealer) should indeed stake a fixed amount to appeal
- I would assume mostly it is about turning submission that judged as invalid to valid
	- if the invalid submission end up valid after the appeal, the warden do not lose the staked fund
	- if the appeal fail, the staked fund are distributed to judge
- the wardne should only appeal with coded POC
